### PR TITLE
Refactor Dockerfile location

### PR DESCRIPTION
**Pull request by michaelkaye**
This is based on the changes in #3253 but old branch is old, so tidying it up and re-submitting as a new, clean PR.

This is step one in a three stage plan:

1) Move docker file around so README from the root of synapse doesn't trample the readme on the docker-hub all the time. We want different information in each place so sharing the same README doesn't seem sensible.
2) Enable automated builds of tags, and develop (only possible as each time you build on docker-hub the readme is updated, so these automated builds trample the README each time, which we re-upload a nice copy of after the manual trigger is complete).
3) Enable automated builds of master to the ":latest" tag which enables people to shoot themselves in the foot by ignoring versioning of their synapse installation and configuration (opinions differ on whether that's a good idea)

### CODE REVIEW

File: contrib/docker/README.md [hunk 1]
```diff
@@ -54,94 +36,6 @@ Then, customize your configuration and run the server:
...
-* `POSTGRES_USER` - The user for the synapse postgres database. [default: `matrix`]
-
-Mail server specific values (will not send emails if not set):
+### More information
 
-* ``SYNAPSE_SMTP_HOST``, hostname to the mail server.
-* ``SYNAPSE_SMTP_PORT``, TCP port for accessing the mail server [default ``25``].
-* ``SYNAPSE_SMTP_USER``, username for authenticating against the mail server if any.
-* ``SYNAPSE_SMTP_PASSWORD``, password for authenticating against the mail server if any.
+For more information on required environment variables and mounts, see the main docker documentation at `docker/README.md`
```
Comment (richvdh) [inline contrib/docker/README.md]: can has link?

File: README.rst [hunk 2]
```diff
@@ -157,7 +157,7 @@ if you prefer.
 
 In case of problems, please see the _`Troubleshooting` section below.
 
-There is an offical synapse image available at https://hub.docker.com/r/matrixdotorg/synapse/tags/ which can be used with the docker-compose file available at `contrib/docker`. Further information on this including configuration options is available in `contrib/docker/README.md`.
+There is an offical synapse image available at https://hub.docker.com/r/matrixdotorg/synapse/tags/ which can be used with the docker-compose file available at `contrib/docker`. Further information on this including configuration options is available in the README on hub.docker.com.
```
Comment (richvdh) [inline README.rst]: can has link?

Comment (richvdh) [inline README.rst]: also, can has wrapping at 80 chars?

File: docker/README.md [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+We instead release only tagged versions so upgrading between releases is entirely
+within your control.
+
+### Using docker-compose (easier)
+
+This image is designed to run either with an automatically generated configuration
+file or with a custom configuration that requires manual editing.
+
+An easy way to make use of this image is via docker-compose. See the
+(https://github.com/matrix-org/synapse/tree/develop/contrib/docker)[contrib]
```
Comment (richvdh) [inline docker/README.md]: that's not how you link in MD. think it's `[text](http://link)`.

File: contrib/docker/README.md [hunk 4]
```diff
@@ -54,94 +36,6 @@ Then, customize your configuration and run the server:
...
-* `POSTGRES_USER` - The user for the synapse postgres database. [default: `matrix`]
-
-Mail server specific values (will not send emails if not set):
+### More information
 
-* ``SYNAPSE_SMTP_HOST``, hostname to the mail server.
-* ``SYNAPSE_SMTP_PORT``, TCP port for accessing the mail server [default ``25``].
-* ``SYNAPSE_SMTP_USER``, username for authenticating against the mail server if any.
-* ``SYNAPSE_SMTP_PASSWORD``, password for authenticating against the mail server if any.
+For more information on required environment variables and mounts, see the main docker documentation at [/docker/README.md](../../docker/README.md)
```
Comment (jcgruenhage) [inline contrib/docker/README.md]: This file is kept for not breaking links I guess?

**Comment by michaelkaye** [comment]
@matrixbot retest this please

**Pull request merged**